external/resinsight/LibCore/cvfVector4.inl

```diff
@@ -491,7 +491,7 @@ inline void Vector4<S>::setZero()
 template<typename S>
 inline bool Vector4<S>::isZero() const
 {
-    return (m_v[0] == 0) && (m_v[1] == 0) && (m_v[2] == 0) && (m_v[2] == 0);
+    return (m_v[0] == 0) && (m_v[1] == 0) && (m_v[2] == 0) && (m_v[3] == 0);
```

Comment:
Yep, that's definitely a bug&mdash;good catch!  Another option would be something like
```C++
return std::all_of(m_v, m_v + (sizeof m_v)/(sizeof m_v[0]), [](const auto& vi) { return vi == 0; });
```
which is slightly more resilient to these kinds of problems.  The downside is that that might not inline quite as well as the current version.